tests/test-umockdev.py

```diff
@@ -215,6 +215,55 @@ def on_uevent(client, action, device, counters):
         mainloop.run()
         self.assertEqual(counter[syspath], [1, 1])
 
+    def test_remove_usb_parent_device(self):
+        '''testbed removing a USB parent device removes children'''
+
+        counter = {}
+
+        def on_uevent(client, action, device, counters):
+            key = device.get_sysfs_path()
+            counters.setdefault(key, [0, 0])
+            if action == 'add':
+                counters[key][0] += 1
+            elif action == 'remove':
+                counters[key][1] += 1
+
+        # set up listener for uevent signal
+        client = GUdev.Client.new(['usb'])
+        client.connect('uevent', on_uevent, counter)
+        mainloop = GLib.MainLoop()
+
+        parent_syspath = self.testbed.add_device('usb', 'myparentdev', None, ['idVendor', '0815'], ['ID_INPUT', '1'])
+        self.assertNotEqual(parent_syspath, None)
+
+        child1_syspath = self.testbed.add_device('usb', 'child1', parent_syspath, ['idVendor', '0815'], ['ID_INPUT', '1'])
+        self.assertNotEqual(child1_syspath, None)
+
+        child2_syspath = self.testbed.add_device('usb', 'child2', parent_syspath, ['idVendor', '0815'], ['ID_INPUT', '1'])
+        self.assertNotEqual(child2_syspath, None)
+
+        # Run the main loop for 0.5 seconds to catch the "add" uevent, which
+        # should be automatically generated by the call to `add_device()`.
+        GLib.timeout_add(500, mainloop.quit)
+        mainloop.run()
+        self.assertIn(parent_syspath, counter)
+        self.assertIn(child1_syspath, counter)
+        self.assertIn(child2_syspath, counter)
+        self.assertEqual(counter[parent_syspath], [1, 0])
+        self.assertEqual(counter[child1_syspath], [1, 0])
+        self.assertEqual(counter[child2_syspath], [1, 0])
+
+        # Just remove the parent, the children should also be removed.
+        self.testbed.remove_device(parent_syspath)
```

Comment:
Can you please add an assertion that the parent syspath is gone from the file system after that? That implies that the children are gone, too.

Reply:
Done.